### PR TITLE
JENKINS-39762 - Update application header

### PR DESCRIPTION
**Pull request by sophistifunk**
* Pull in updated JDL, and add new components to replace existing headers.
 * Add Blue Ocean logo component to use for betas.
 * Add storybook entries to demonstrate usage.

The existing stuff hasn't been replaced, as Thor's doing stuff with the headers, but mostly I ran out of steam and 2016. It's new components, which should be demonstrated amply in Storybooks.

Squashed commit of the following:

    josh/JENKINS-39762-new-headers * lint
    josh/JENKINS-39762-new-headers * ResultPageHeader working nicely, no build details shown in storybooks
    josh/JENKINS-39762-new-headers * Progess on ResultPageHeader, still losing result somewhere :(
    josh/JENKINS-39762-new-headers * Improvments to ContentPageHeader, show weather icon in story
    josh/JENKINS-39762-new-headers * Fix JDL version
    josh/JENKINS-39762-new-headers * Clean up the rest of the storybook remnants from web
    josh/JENKINS-39762-new-headers * Move components and stories to -dashboard
    josh/JENKINS-39762-new-headers * ehsuydjmntws5rty67hauw5r6yjthna
    josh/JENKINS-39762-new-headers * tweak storybook to get by with a npm run bundle rather than full mvn build which is way slower and requirequires lint
    josh/JENKINS-39762-new-headers * Add logo as a component, rejigger storybooks to also reference module-local styles
    josh/JENKINS-39762-new-headers * Add storybooks to -web
    josh/JENKINS-39762-new-headers * Pull in JDL beta, switch out PageTabs for PageTabsOld for transition

# Description

See [JENKINS-39762](https://issues.jenkins-ci.org/browse/JENKINS-39762).
See [JENKINS-39763](https://issues.jenkins-ci.org/browse/JENKINS-39763).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [X] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by sophistifunk** [comment]
Taste the rainbow

resultheaders

**Comment by sophistifunk** [comment]
screen shot 2016-12-22 at 5 21 02 pm

**Comment by michaelneale** [comment]
very nice - @scherler @i386 want to take a look at storybooks?

**Comment by i386** [comment]
@sophistifunk this is looking really nice mate.

**Comment by michaelneale** [comment]
cc @cliffmeyers code review possible ?

File: blueocean-dashboard/src/main/js/components/RunDetails.jsx [hunk 1]
```diff
@@ -187,7 +187,7 @@ class RunDetails extends Component {
                             <TabLink to="/artifacts">{t('rundetail.header.tab.artifacts', {
                                 defaultValue: 'Artifacts',
                             })}</TabLink>
-                        </PageTabs>
+                        </PageTabsOld>
```
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/components/RunDetails.jsx]: Just confirming: `RunDetails` will continue to use the old tab style for now?

Comment (sophistifunk) [inline blueocean-dashboard/src/main/js/components/RunDetails.jsx]: I did a full search/replace for now, so no code except the *new* header components should be using the new `PageTabs`, and once the components have been merged into actual pages in place of the old headers we'll get rid of `PageTabsOld` from JDL completely.

**Comment by sophistifunk** [comment]
I'll rebase this and clean-build and give everything a manual once-over today.

**Pull request merged**